cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -170,8 +170,8 @@ public void deleteNamespacePrefs(HttpRequest request, HttpResponder responder,
   @GET
   public void getAppPrefs(HttpRequest request, HttpResponder responder,
                           @PathParam("namespace-id") String namespace, @PathParam("application-id") String appId,
-                          @QueryParam("resolved") boolean resolved) throws Exception {
```

Comment:
should we keep the exception?

Reply:
But my IDE says it's no throw

Reply:
interesting, so what happens if appId does not exist in db?